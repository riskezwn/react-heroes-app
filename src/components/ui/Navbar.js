import React from "react";
import { Link, NavLink } from "react-router-dom";

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <ul className="navbar-nav">
                    <Link className="navbar-brand" to="/">
                        HeroesApp
                    </Link>
                    <li className="nav-item">
                        <NavLink
                            className="nav-link"
                            aria-current="page"
                            to="/marvel"
                        >
                            Marvel
                        </NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink className="nav-link" to="/dc">
                            DC
                        </NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink className="nav-link" to="/search">
                            Search
                        </NavLink>
                    </li>
                </ul>
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink className="nav-link" to="/login">
                            Logout
                        </NavLink>
                    </li>
                </ul>
            </div>
        </nav>
    );
};
