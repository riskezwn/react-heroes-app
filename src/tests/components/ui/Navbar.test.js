import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";
import "../../../setupTest";
import { act } from "react-dom/cjs/react-dom-test-utils.development";

describe("test en <Navbar/>", () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: "riskezwn",
        },
    };
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
        </AuthContext.Provider>
    );
    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(".text-info").text().trim()).toBe("riskezwn");
    });

    test("debe de llamar el logout y usar Navigate", () => {
        act(() => {
            wrapper.find("button").prop("onClick")();
        });

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout,
        });
    });
});
