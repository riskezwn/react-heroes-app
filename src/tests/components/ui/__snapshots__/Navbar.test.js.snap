// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en <Navbar/> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <ul
      className="navbar-nav"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        HeroesApp
      </a>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </li>
    </ul>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item nav-link text-info"
      >
        riskezwn
      </li>
      <li
        className="nav-item"
      >
        <button
          className="nav-link btn"
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
